### PR TITLE
Hollow/boid health

### PR DESCRIPTION
**Pull request by abbclas**

### CODE REVIEW

File: app/Scripts/boid.gd [hunk 1]
```diff
@@ -13,26 +13,58 @@ export var follow_target = Vector2()
...
 	self.set_meta("Boid", false)
 	get_parent().get_node("Enemy").connect("_enemy_moused_over_true", self, "_enemy_moused_over_true")
 	get_parent().get_node("Enemy").connect("_enemy_moused_over_false", self, "_enemy_moused_over_false")
+	
 func _process(delta):
 	#find all boids in the normal process to keep accurate track of them, all
 	#other calculations will be done in physics process to keep them framerate independent
 	boids = find_all_boids()
-
+	HealthCalculations()
```
Comment (noh0st) [inline app/Scripts/boid.gd]: we don' need to run this every frame, just at initialization and when damage is taken right?

File: app/Scripts/boid.gd [hunk 2]
```diff
@@ -71,7 +103,7 @@ func find_all_boids():
 func cohesion():
 	var percieved_center = Vector2(0,0)
 	for b in boids:
-		if(b!=self):
+		if(b!=self && boids.has(boids.find(b)) ):
```
Comment (noh0st) [inline app/Scripts/boid.gd]: what is `boids.has(boids.find(b)` doing? We're looping through boids, so why isn't `b` already known to be in `boids`?

**Comment by noh0st** [comment]
seems like number of followers doesn't always match number of boids following?

File: app/Scripts/boid.gd (hunk 2 above)
Comment (abbclas) [reply app/Scripts/boid.gd]: Yep Idk why that happens and for some reason I need to check to see if the boid exist or not, If I don't the script will try to access a dead boid for some reason and gives me errors

File: app/Scripts/boid.gd (hunk 1 above)
Comment (abbclas) [reply app/Scripts/boid.gd]: Yes but for some reason setget wasnt working

**Comment by abbclas** [comment]
> seems like number of followers doesn't always match number of boids following?

When does that happen?

File: app/Scripts/boid.gd (hunk 1 above)
Comment (noh0st) [reply app/Scripts/boid.gd]: do you want to figure out why it wasn't working?

File: app/Scripts/boid.gd (hunk 2 above)
Comment (noh0st) [reply app/Scripts/boid.gd]: Can we figure out why?

**Comment by noh0st** [comment]
> > seems like number of followers doesn't always match number of boids following?
> 
> When does that happen?

They may be overlapping eachother actually so it appears the count if off. Boids are supposed to have a separation force so that shouldn't happen - on main they do not overlap. So seems like these changes are messing with the boid behavior somehow

**Review by noh0st — APPROVED** [review]
we can merge this. Right now each boid itself needs to keep track of the boids array, which doesn't seem necessary? We should think about how to solve that going forward

**Pull request merged**